### ISSUE TITLE
Serializer Specific Caching

### ISSUE THREAD

**Issue by kpheasey** [fast-jsonapi/fast_jsonapi#71]
## Expected Behavior

Passing one record to multiple serializers should result in 2 separate cached responses.
ProjectSerializer
ProjectListingSerializer

## Actual Behavior

Cache only returns the first serialized instance of the record, regardless of what serializer is being called.

ProjectSerializer is called first
ProjectListingSerializer returns ProjectSerializer response
## Steps to Reproduce the Problem

  1. Create SerializerA and SerializerB
  2. Pass SerializerA record A
  3. Pass SerializerB record A

## Proposed Solution

The cache key should include the serializer name by default.  If I create two separate serializers with separate attribute and relationship sets, the result of one cache should not be shared with the other.  Caching should happen at a Model + Serializer level.

If there is no opposition, I will work on the solution shortly.

**Comment by stas** [comment]
@kpheasey I'm looking at this and I'm not sure this is a bug.

Starting with the #51 the cache options support the [`namespace` and `version`](https://github.com/fast-jsonapi/fast_jsonapi#caching). These both can be configured to use the serializer name. Consider:
```
cache_options store: Rails.cache, namespace: 'fast-jsonapi.<YOUR SERIALIZER TYPE>', expires_in: 1.hour
```

Could you confirm this is solving your issue?

**Comment by stas** [comment]
This should be fixed as of #51
